extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,26 +1,73 @@
-import {Layout, applyFillContent} from '#core/dom/layout';
+import {Layout} from '#core/dom/layout';
 
-const TAG = 'amp-story-shopping-config';
+import {Services} from '#service';
+
+import {devAssert} from '#utils/log';
+
+import {
+  Action,
+  ShoppingConfigDataDef,
+} from '../../amp-story/1.0/amp-story-store-service';
+
+/**
+ * @typedef {{
+ *  items: !Array<!ShoppingConfigDataDef>,
+ * }}
+ */
+let ShoppingConfigDef;
 
 export class AmpStoryShoppingConfig extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
-    /** @private {string} */
-    this.myText_ = TAG;
+    /** @private @const {?./amp-story-request-service.AmpStoryRequestService} */
+    this.requestService_ = null;
+
+    /** @private @const {?./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+  }
+
+  /**
+   * Keys product data to product-ids and adds them to the store service.
+   * @param {!ShoppingConfigDef} storyConfig
+   * @private
+   */
+  addShoppingDataFromConfig_(storyConfig) {
+    const productIDtoProduct = {};
+
+    for (const item of storyConfig['items']) {
+      productIDtoProduct[item['product-tag-id']] = item;
+    }
+
+    this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, productIDtoProduct);
 
-    /** @private {?Element} */
-    this.container_ = null;
+    //TODO(#36412): Add call to validate config here.
   }
 
   /** @override */
   buildCallback() {
     super.buildCallback();
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    applyFillContent(this.container_, /* replacedContent */ true);
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win),
+      Services.storyRequestServiceForOrNull(this.win),
+    ])
+      .then((services) => {
+        this.storeService_ = services[0];
+        this.requestService_ = services[1];
```

Comment:
suggestion:
We can unpack the array in the argument. This is a stylistic choice so it's up to you if you want to use it!
      .then(([storeService, requestService]) => {
        this.storeService_ = storeService;
        this.requestService_ = requestService;

Reply:
I like it! Let's go with this.